pkg/obc/obc.go

```diff
@@ -435,3 +495,80 @@ func CheckPhase(obc *nbv1.ObjectBucketClaim) {
 		log.Printf("⏳ OBC %q Phase is %q", obc.Name, obc.Status.Phase)
 	}
 }
+
+// GenerateAccountKeys regenerate noobaa OBC account S3 keys
+func GenerateAccountKeys(name string, bucketName string) error {
+	log := util.Logger()
+
+	var accessKeys nb.S3AccessKeys
+
+	sysClient, err := system.Connect(true)
+	if err != nil {
+		return err
+	}
+
+	// Checking that we can find the secret before we are calling the RPC to change the credentials.
+	secret := util.KubeObject(bundle.File_deploy_internal_secret_empty_yaml).(*corev1.Secret)
+	secret.Namespace = options.Namespace
+	secret.Name = name
+	if !util.KubeCheckQuiet(secret) {
+		log.Fatalf(`❌  Could not find secret: %s, will not regenerate keys.`, secret.Name)
+	}
+
+	nbClient := sysClient.NBClient
+	list, err := nbClient.ListAccountsAPI(nb.ListAccountsParams{	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if len(list.Accounts) == 0 {
+		log.Fatalf(`❌  No accounts found.\n`)
+	}
+
+	AccountName := ""
+	for k := range list.Accounts {
+		if strings.Contains(list.Accounts[k].Email, bucketName) {
```

Comment:
That's a dangerous approach for checking if this is the account we should regenerate..
for example, if the current account's email in the loop is ```obc-account.obc1obc1.%x@noobaa.io``` and the bucketName is obc1, you would change another account that you are not supposed to...